### DISCUSSION TITLE
Create new Value from multiple Values

### DISCUSSION THREAD

**Discussion by Tommyten** [arkivanov/Decompose#97] [Ideas]
It would be great to have a function to create a new Value from several Values, kind of like Compose's `derivedStateOf()`. I know that there is map, this however takes in only one Value, or am I overlooking something?

Use case could be for example:
You have a parent Component with several ChildComponents. When at least one of the ChildComponents is loading something, set a Value `loading` of the parent component to true, to show a Loading Indicator.

**Comment by arkivanov** [comment]
Thanks for the use case! At the moment I'm trying to avoid additional operators for Value, to avoid creating a yet-another-rx-library. Maybe instead you can expose a `StateFlow` or Reaktive's `BehavourObservable` (depends on what you use)?

**Comment by Tommyten** [comment]
Thanks for the suggestion! I'll probably resort to using `StateFlow`.
